File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        res = []
        l,r = 0, len(numbers)-1
        
        while l < r:
            complement = target - numbers[l]
            if complement > numbers[r]:
                l +=1
            elif complement < numbers[r]:
                r -= 1
            else:
                return [l+1,r+1]

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> index;
        for(int i = 0; i < nums.size(); i++){
            int comp = target - nums[i];
            if(index.contains(comp)){
                return {index[comp], i};
            }
            index[nums[i]]=i;
        }
    }
};
